parsley-debug/jvm/src/main/scala/parsley/debugger/util/XCollectorJava.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Parsley Contributors <https://github.com/j-mie6/Parsley/graphs/contributors>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package parsley.debugger.util
+
+import scala.collection.mutable
+import scala.util.{Success, Try}
+
+import parsley.debugger.internal.Rename.MapAddAll
+import parsley.token.Lexer
+
+import parsley.internal.deepembedding.frontend.LazyParsley
+
+// Java-reflection powered XCollector. For backup purposes. Not meant to be used directly, but inherited from.
+// The setAccessible methods on private members is a deprecated action, so proceed with caution.
+// $COVERAGE-OFF$
+private [parsley] abstract class XCollectorJava extends CollectorImpl {
+    override val supported: Boolean = true
+
+    private def getParsers(obj: Any): Try[Map[LazyParsley[_], String]] = {
+        val accumulator: mutable.HashMap[LazyParsley[_], String] = new mutable.HashMap()
+
+        Try {
+            // FIXME: Type matching by name is very brittle...
+            val parserGetters = obj.getClass.getDeclaredMethods.filter { mth =>
+                Seq(Class.forName("parsley.Parsley"), Class.forName("parsley.internal.deepembedding.frontend.LazyParsley"))
```

Comment:
Perhaps a comment in these two classes pointing out a change needed here if they change

Reply:
I think changes internally to `Parsley` or `LazyParsley` won't break this. It'd only be if they change package path for some reason (e.g. `parsley.Parsley` becoming something like `parsley.parsers.Parsley`).

Reply:
Yes, exactly, which is why a comment there wouldn't be such a bad idea 🙂. Though I don't expect I'll touch the package structure from now on

Reply:
I could probably even automate this by one of the methods defined in `Class[_]` to automatically get the package path when required.